tool/net/redbean.c

```diff
@@ -4697,6 +4697,162 @@ static int LuaGetHttpReason(lua_State *L) {
   return 1;
 }
 
+static int EncodeJsonData(lua_State *L, char **buf, int level, char *numformat) {
+  size_t idx = -1;
+  size_t tbllen, buflen;
+  bool isarray;
+  int t = lua_type(L, idx);
+  if (level < 0) return luaL_argerror(L, 1, "too many nested tables");
+  if (LUA_TSTRING == t) {
+    appendf(buf, "\"%s\"", gc(EscapeJsStringLiteral(lua_tostring(L, idx), -1, 0)));
+  } else if (LUA_TNUMBER == t) {
+    // TODO: what if int64_t isn't representable as double
```

Comment:
Two possible options. One is make it NAN. The other is raise an error. Saturation clamping we could do but it wouldn't feel right in the general case. It'd help to know what other json impls do.

Reply:
NAN is not acceptable in (standard) JSON, although there are some JSON implementation that may use it. Raising an error should be fine. What's a good/fast way to check for that condition?

Reply:
@jart, should I simply compare the integer with `0x7FEFFFFFFFFFFFFF` per this SO answer: https://stackoverflow.com/a/39783981/1442917?

Reply:
Got it; based on your other comment I removed the TODO comments to keep things clear.

Reply:
Honestly I think for a general purpose abstraction like JSON the right thing to do is probably to do nothing. See comment I left in the other thread. Maybe leave a comment saying `/* precision loss possible */` in case we ever find reason to make the behavior configurable. Please note undefined behavior has the following definition per ANSI C standard:

```
 * Undefined behavior --- behavior, upon use of a nonportable or
   erroneous program construct, of erroneous data, or of
   indeterminately-valued objects, for which the Standard imposes no
   requirements.  Permissible undefined behavior ranges from ignoring the
   situation completely with unpredictable results, to behaving during
   translation or program execution in a documented manner characteristic
   of the environment (with or without the issuance of a diagnostic
   message), to terminating a translation or execution (with the issuance
   of a diagnostic message).
```

So it would be legal to raise an error if you want. But whatever you choose to do, it's probably worth noting in the docs.